### ISSUE TITLE
Scalar tensor dimension

### ISSUE THREAD

**Issue by ailzhang** [pytorch/xla#1958]
This is more like one edge case shared by multiple ops (found when running test_autograd.py). 
Each line of `print` below triggers the same dim out of boundary error, while all passes on CPU. 
```
import torch
import torch_xla

device='xla'
# device='cpu'
a = torch.tensor(2.1, device=device)
print(torch.cumprod(a, 0))
print(torch.cumsum(a, 0))
print(torch.gather(a, 0, torch.tensor(0, device=device, dtype=torch.int64)))
print(torch.index_add(a, 0, torch.tensor(0, device=device, dtype=torch.int64), a))
print(torch.index_copy(a, 0, torch.tensor(0, device=device, dtype=torch.int64), a))
print(torch.index_fill(a, 0, torch.tensor(0, device=device, dtype=torch.int64), 2))
print(torch.index_select(a, 0, torch.tensor(0, device=device, dtype=torch.int64)))
print(torch.kthvalue(a, 1))
print(torch.logsumexp(a, 0))
print(torch.max(a, 0))
print(torch.mean(a, 0))
print(torch.min(a, 0))
print(torch.norm(a, 2, 0))
print(torch.prod(a, 0))
print(torch.scatter_add(a, 0, torch.tensor(0, dtype=torch.int64, device=device), a))
print(torch.scatter(a, 0, torch.tensor(0, dtype=torch.int64, device=device), a))
print(torch.squeeze(a, 0))
print(torch.sum(a, 0))
print(torch.topk(a, 1))
print(torch.transpose(a, 0, 0))
```

**Comment by JackCaoG** [comment]
similar issue as https://github.com/pytorch/xla/issues/1960, xla doesn't play well with accessing the dimension 0 of a scalar tensor.
